internal/services/storage/storage_container_resource.go

```diff
@@ -189,31 +206,102 @@ func flattenStorageContainerAccessLevel(input containers.AccessLevel) string {
 	}
 
 	if err := client.Delete(ctx, account.ResourceGroup, id.AccountName, id.Name); err != nil {
-		return fmt.Errorf("deleting Container %q (Storage Account %q / Resource Group %q): %s", id.Name, id.AccountName, account.ResourceGroup, err)
+		return diag.Errorf("deleting Container %q (Storage Account %q / Resource Group %q): %s", id.Name, id.AccountName, account.ResourceGroup, err)
 	}
 
 	return nil
 }
+
+func resourceStorageContainerUpdate(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) diag.Diagnostics {
+	storageClient := meta.(*clients.Client).Storage
+	_ = ctx
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.StorageContainerDataPlaneID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	account, err := storageClient.FindAccount(ctx, id.AccountName)
+	if err != nil {
+		return diag.Errorf("retrieving Account %q for Container %q: %s", id.AccountName, id.Name, err)
+	}
+	if account == nil {
+		return diag.Errorf("Unable to locate Storage Account %q!", id.AccountName)
+	}
+	client, err := storageClient.ContainersClient(ctx, *account)
+	if err != nil {
+		return diag.Errorf("building Containers Client for Storage Account %q (Resource Group %q): %s", id.AccountName, account.ResourceGroup, err)
+	}
+
+	if d.HasChange("container_access_type") {
+		log.Printf("[DEBUG] Updating the Access Control for Container %q (Storage Account %q / Resource Group %q)..", id.Name, id.AccountName, account.ResourceGroup)
+		accessLevelRaw := d.Get("container_access_type").(string)
+		accessLevel := expandStorageContainerAccessLevel(accessLevelRaw)
+
+		if err := client.UpdateAccessLevel(ctx, account.ResourceGroup, id.AccountName, id.Name, accessLevel); err != nil {
+			return diag.Errorf("updating the Access Control for Container %q (Storage Account %q / Resource Group %q): %s", id.Name, id.AccountName, account.ResourceGroup, err)
+		}
+
+		log.Printf("[DEBUG] Updated the Access Control for Container %q (Storage Account %q / Resource Group %q)", id.Name, id.AccountName, account.ResourceGroup)
+	}
+
+	if d.HasChange("metadata") {
+		log.Printf("[DEBUG] Updating the MetaData for Container %q (Storage Account %q / Resource Group %q)..", id.Name, id.AccountName, account.ResourceGroup)
+		metaDataRaw := d.Get("metadata").(map[string]interface{})
+		metaData := ExpandMetaData(metaDataRaw)
+
+		if err := client.UpdateMetaData(ctx, account.ResourceGroup, id.AccountName, id.Name, metaData); err != nil {
+			return diag.Errorf("updating the MetaData for Container %q (Storage Account %q / Resource Group %q): %s", id.Name, id.AccountName, account.ResourceGroup, err)
+		}
+
+		log.Printf("[DEBUG] Updated the MetaData for Container %q (Storage Account %q / Resource Group %q)", id.Name, id.AccountName, account.ResourceGroup)
+	}
+
+	return resourceStorageContainerRead(ctx, d, meta)
+}
+
+func ExpandMetaData(input map[string]interface{}) map[string]string {
```

Comment:
You should have added the file of `metadata.go` in validate folder just like the rest or azurerm since the structure is very similar, instead of adding a function in the resource

Reply:
ready!